### PR TITLE
Use ENV variable to distinguish between self-hosted and managed instance

### PR DESCRIPTION
**Pull request by SailReal**
When deploying Hub, we can now set `HUB_MANAGED_INSTANCE=true` (default is `false`) to indicate if this is a self-hosted or a managed instance which in turn has an influence on the community license: managed instances does not have one.

Unfortunately I didn't get it running to have a test with a custom test profile inside a test with the default profile, that is the reason for the `BillingResourceManagedInstanceTest` class. Sad but wanted to test this case as well.

### CODE REVIEW

File: backend/src/main/java/org/cryptomator/hub/license/LicenseHolder.java [hunk 1]
```diff
@@ -69,7 +75,13 @@ public DecodedJWT get() {
...
 		return Optional.ofNullable(license) //
 				.map(l -> l.getExpiresAt().toInstant().isBefore(Instant.now())) //
-				.orElse(CommunityLicenseConstants.IS_EXPIRED);
+				.orElseGet(() -> {
+					if (!managedInstance) {
+						return CommunityLicenseConstants.IS_EXPIRED;
+					} else {
+						return ManagedInstanceNoLicenseConstants.IS_EXPIRED;
+					}
+				});
```
Comment (tobihagemann) [inline backend/src/main/java/org/cryptomator/hub/license/LicenseHolder.java:84]: I wonder if this is really necessary? If there is no license, it's always "not expired", right? Can't we just say `.orElse(false)`?

File: backend/src/main/java/org/cryptomator/hub/license/LicenseHolder.java [hunk 2]
```diff
@@ -81,7 +93,19 @@ public long getAvailableSeats() {
...
+				.orElseGet(this::getNoLicenseSeats);
+	}
+
+	public long getNoLicenseSeats() {
+		if (!managedInstance) {
+			return CommunityLicenseConstants.SEATS;
+		} else {
+			return ManagedInstanceNoLicenseConstants.SEATS;
+		}
+	}
```
Comment (tobihagemann) [inline backend/src/main/java/org/cryptomator/hub/license/LicenseHolder.java:105]: How about renaming `CommunityLicenseConstants` to `SelfHostedNoLicenseConstants`? And also remove `Instance` in `ManagedInstanceNoLicenseConstants`?

File: frontend/src/i18n/en-US.json [hunk 3]
```diff
@@ -37,9 +37,11 @@
   "admin.licenseInfo.expiresAt.description.valid": "Your license is valid.",
   "admin.licenseInfo.expiresAt.description.expired": "Your license has expired.",
   "admin.licenseInfo.manageSubscription": "Manage Subscription",
+  "admin.licenseInfo.type.title": "Type",
+  "admin.licenseInfo.upgradeLicense": "Get License",
```
Comment (tobihagemann) [inline frontend/src/i18n/en-US.json:41]: Rename `admin.licenseInfo.upgradeLicense` to `admin.licenseInfo.getLicense`?

File: frontend/src/i18n/en-US.json [hunk 4]
```diff
@@ -37,9 +37,11 @@
...
   "admin.licenseInfo.communityLicense.description": "Thank you for using Cryptomator Hub! You have been granted the Community License. If you need more seats, upgrade your license.",
```
Comment (tobihagemann) [inline frontend/src/i18n/en-US.json:42]: Rename `admin.licenseInfo.communityLicense.description` to `admin.licenseInfo.selfHostedNoLicense.description`?

File: backend/src/main/java/org/cryptomator/hub/license/LicenseHolder.java (hunk 1 above)
Comment (SailReal) [reply backend/src/main/java/org/cryptomator/hub/license/LicenseHolder.java:84]: Right, firstly Managed wasn't expired and after changing, I missed to change this block back, thx!

Comment (SailReal) [reply backend/src/main/java/org/cryptomator/hub/license/LicenseHolder.java]: Fixed in https://github.com/cryptomator/hub/pull/174/commits/9a7e1d46f452775cccec5351b8291a600f66f10e

File: frontend/src/i18n/en-US.json (hunk 3 above)
Comment (SailReal) [reply frontend/src/i18n/en-US.json]: Fixed in https://github.com/cryptomator/hub/pull/174/commits/9a7e1d46f452775cccec5351b8291a600f66f10e

File: frontend/src/i18n/en-US.json (hunk 4 above)
Comment (SailReal) [reply frontend/src/i18n/en-US.json]: Fixed in https://github.com/cryptomator/hub/pull/174/commits/9a7e1d46f452775cccec5351b8291a600f66f10e

**Pull request merged**